cl/citations/management/commands/cl_find_citations_recap.py

```diff
@@ -0,0 +1,180 @@
+            type=valid_date_time,
+            help="End date in ISO-8601 format for a range of documents to "
+            "update.",
+        )
+        parser.add_argument(
+            "--modified-after",
+            type=valid_date_time,
+            help="The modification date ISO-8601 format for a range of "
+            "documents to update.",
+        )
+        parser.add_argument(
+            "--all",
+            action="store_true",
+            default=False,
+            help="Parse citations for all stored RECAP documents",
+        )
+        parser.add_argument(
+            "--queue",
+            default="batch1",
+            help="The celery queue where the tasks should be processed.",
+        )
+
+        def update_documents(self, pks: Iterable, queue_name: str):
+            sys.stdout.write(f"Graph size is {self.count:d} nodes.\n")
+            sys.stdout.flush()
+
+            chunk = []
+            chunk_size = 100
+            processed_count = 0
+            throttle = CeleryThrottle(queue_name=queue_name)
+
+            for pk in pks:
+                throttle.maybe_wait()
+                processed_count += 1
+                last_item = self.count == processed_count
+                chunk.append(pk)
+
+                if processed_count % chunk_size == 0 or last_item:
+                    find_citations_and_parantheticals_for_recap_documents.apply_async(
+                        args=(chunk),
+                        queue=queue_name,
+                    )
+                    chunk = []
+
+                self.log_progress(processed_count, pk)
+
+        def handle(self, *args: List[str], **options: OptionsType):
+            super(Command, self).handle(*args, **options)
+            both_list_and_endpoints = options.get("doc_id") is not None and (
+                options.get("start_id") is not None
+                or options.get("end_id") is not None
+                or options.get("filed_after") is not None
+                or options.get("filed_before") is not None
+                or options.get("modified_after") is not None
+            )
+            no_option = not any(
+                [
+                    options.get("doc_id") is None,
+                    options.get("start_id") is None,
+                    options.get("end_id") is None,
+                    options.get("filed_after") is None,
+                    options.get("filed_before") is None,
+                    options.get("modified_after") is None,
+                    options.get("all") is False,
+                ]
+            )
+            if both_list_and_endpoints or no_option:
+                raise CommandError(
+                    "Please specify either a list of documents, a "
+                    "range of ids, a range of dates, or "
+                    "everything."
+                )
+
+            self.index = options["index"]
+            query = RECAPDocument.objects.all().order_by("pk")
```

Comment:
Added `is_available=True` filter to the base query. 

I also removed the extra logic for handling the "all", because it's actually slightly worse than redundant, I think. I could be wrong, but doesn't `Model.objects.all()` load the entire table into memory? But then, this is done only to subsequently call `values_list` to get an iterator of PKs. 

I tested the new flow in the Django shell, and it appears to work.

Reply:
Well, Django only runs queries when you access their underlying data, so just doing a call to `.all()` doesn't touch the DB. It's only when you access it by printing it, iterating it, slicing it, whatever.

But yeah, if it's redundant, no point in it being here.